pennylane/beta/tapes/tape.py

```diff
@@ -890,6 +906,60 @@ def execute(self, device, params=None):
 
         return self._execute(params, device=device)
 
+    def _get_all_parameters(self, params):
+        """Return all parameters by combining trainable parameters supplied by ``params`` with
+        existing non-trainable parameters.
+
+        The returned parameters are provided in order of appearance
+        on the tape.
+
+        Args:
+            params (list[Any]): The quantum tape operation parameters.
+
+        **Example**
+
+        .. code-block:: python
+
+            from pennylane.beta.tapes import QuantumTape
+            from pennylane.beta.queuing import expval, var, sample, probs
+
+            with QuantumTape() as tape:
+                qml.RX(0.432, wires=0)
+                qml.RY(0.543, wires=0)
+                qml.CNOT(wires=[0, 'a'])
+                qml.RX(0.133, wires='a')
+                expval(qml.PauliZ(wires=[0]))
+
+        Suppose only parameters 0 and 2 are trainable:
+
+        >>> tape.trainable_params = {0, 2}
+
+        We can access all parameters using:
+
+        >>> tape._get_all_parameters([0.1, 0.2])
+        [0.1, 0.543, 0.2]
+        """
+        num_all_parameters = len(self._par_info)  # including non-trainable parameters
+
+        if self.num_params == num_all_parameters:
+            return params
+        # Otherwise, we must combine the trainable parameters supplied by the params
+        # argument with the non-trainable parameters given by get_parameters()
+
+        saved_all_parameters = self.get_parameters(trainable_only=False)
+
+        all_parameters = []
+        position = 0
+        for i in range(num_all_parameters):
+            if i in self._trainable_params:
+                p = params[position]
+                position += 1
```

Comment:
What happens if there aren't enough parameters passed for `params`? E.g. getting an error for `tape._get_all_parameters([0.1])`. If we'd like to use the already supplied parameters for the trainable parameters, then maybe it's worth extending the condition with `position<=num_params`.
```suggestion
        num_params = len(params)
        for i in range(num_all_parameters):
            if i in self._trainable_params and position<=num_params:
                p = params[position]
                position += 1
```